src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -126,21 +129,26 @@ public long getFlushTimeThresholdSeconds() {
    * @param ingestionMethodConfig ingestion method used. Check {@link IngestionMethodConfig}
    * @return true if all thresholds are valid.
    */
-  public static boolean validateBufferThreshold(
+  public static Set<String> validateBufferThreshold(
```

Comment:
the three verify* methods in this class are very similar. can i refactor them into one method?

Reply:
Sure! Can you rename the methods too since it is not only verifying but also returning? and change javadocs for returning a set instead of a boolean now?

Reply:
The reason I kept them separate was because of those methods had different behaviors to verify.